### PR TITLE
Added the time changing for notification feature

### PR DESCRIPTION
**Pull request by ashishsaranshakya**
as mentioned in the issues section of the repo the feature works as i have tested it on 2 devices in real time environment.

Still I would like to ask the owner to try and somehow set exact alarm as it is not being offered as a built in function of the alarm manager class

Minor changes to UI side of things might be required aa the owner wants

### CODE REVIEW

**Comment by ashishsaranshakya** [comment]
lastly can I get your contact info, would like to meet you

File: app/src/main/java/tk/therealsuji/vtopchennai/activities/MainActivity.java [hunk 1]
```diff
@@ -417,4 +419,11 @@ protected void onDestroy() {
         super.onDestroy();
         compositeDisposable.dispose();
     }
+
+    private void addNotificationInterval(){
+        SharedPreferences sharedPreferences=SettingsRepository.getSharedPreferences(this);
+        if (!sharedPreferences.contains("notification_interval")){
+            sharedPreferences.edit().putInt("notification_interval",30).apply();
+        }
+    }
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/activities/MainActivity.java:428]: I don't see why the changes in this file is required, 30 is taken as the default value anyway.

File: app/src/main/java/tk/therealsuji/vtopchennai/fragments/ProfileFragment.java [hunk 2]
```diff
@@ -146,6 +161,61 @@ public class ProfileFragment extends Fragment {
...
+                    context -> {
+
+                        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(context);
+                        builder.setTitle("Set Notification Timing");
+                        View viewInflated = LayoutInflater.from(getContext()).inflate(R.layout.frame_notification_interval, (ViewGroup) getView(), false);
+                        final EditText input = viewInflated.findViewById(R.id.input);
+                        ImageView image=viewInflated.findViewById(R.id.image_info);
+                        SharedPreferences sharedPreferences=SettingsRepository.getSharedPreferences(context);
+                        if (getTheme(getContext())==THEME_DAY || getTheme(getContext())==THEME_SYSTEM_DAY) image.setImageDrawable(context.getDrawable(R.drawable.ic_info_dark));
+                        else image.setImageDrawable(context.getDrawable(R.drawable.ic_info_light));
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/fragments/ProfileFragment.java:177]: It's a waste of space to have multiple drawables for different themes, a single drawable with pre-defined colors can be used.

File: app/src/main/res/layout/frame_notification_interval.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            tools:ignore="Autofill,HardcodedText" />
+
+        <ImageView
+            android:id="@+id/image_info"
+            android:paddingVertical="10dp"
+            android:layout_marginTop="10dp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:contentDescription="Info icon"
+            tools:ignore="HardcodedText" />
```
Comment (therealsujitk) [inline app/src/main/res/layout/frame_notification_interval.xml:29]: In general, it isn't a great idea to ignore warnings (unless in rare cases where it can't be helped). For this one in particular, having the strings in `strings.xml` improves the overall app accessibility.

**Review by therealsujitk — COMMENTED** [review]
I would prefer having a dialog with some predefined options (similar to the semester selection dialog) instead of allowing the user to set a custom value.

File: app/src/main/java/tk/therealsuji/vtopchennai/fragments/ProfileFragment.java (hunk 2 above)
Comment (ashishsaranshakya) [reply app/src/main/java/tk/therealsuji/vtopchennai/fragments/ProfileFragment.java:177]: i dont know how to do that

**Review by therealsujitk — COMMENTED** [review]
Most of these changes are irrelevant, unnecessary and students will have several problems updating to this version, therefore I will be closing this PR.
